rllib/core/optim/tests/test_rl_optimizer_tf.py

```diff
@@ -0,0 +1,477 @@
+    env: gym.Env, module_for_inference: RLModule, num_rollouts: int
+) -> SampleBatch:
+    _returns = []
+    _rollouts = []
+    for _ in range(num_rollouts):
+        obs = env.reset()
+        _obs, _next_obs, _actions, _rewards, _dones = [], [], [], [], []
+        _return = -0
+        for _ in range(env._max_episode_steps):
+            _obs.append(obs)
+            fwd_out = module_for_inference.forward_inference(
+                {"obs": tf.convert_to_tensor([obs], dtype=tf.float32)}
+            )
+            action = convert_to_numpy(fwd_out["action_dist"].sample())[0]
+            next_obs, reward, done, _ = env.step(action)
+            _next_obs.append(next_obs)
+            _actions.append([action])
+            _rewards.append([reward])
+            _dones.append([done])
+            _return += reward
+            if done:
+                break
+            obs = next_obs
+        batch = SampleBatch(
+            {
+                "obs": _obs,
+                "next_obs": _next_obs,
+                "actions": _actions,
+                "rewards": _rewards,
+                "dones": _dones,
+            }
+        )
+        _returns.append(_return)
+        _rollouts.append(batch)
+    return np.mean(_returns), _returns, _rollouts
+
+
+@dataclass
+class FCConfig:
+    """Configuration for a fully connected network.
+
+    Attributes:
+        input_dim: The input dimension of the network. It cannot be None.
+        output_dim: The output dimension of the network. if None, the last layer would
+            be the last hidden layer.
+        hidden_layers: The sizes of the hidden layers.
+        activation: The activation function to use after each layer (except for the
+            output).
+        output_activation: The activation function to use for the output layer.
+    """
+
+    input_dim: int = None
+    output_dim: int = None
+    hidden_layers: List[int] = field(default_factory=lambda: [256, 256])
+    activation: tf.keras.layers.Layer = tf.keras.layers.ReLU
+    output_activation: str = None
+
+
+class TFFCNet(tf.keras.Model):
+    """A simple fully connected network.
+
+    Attributes:
+        input_dim: The input dimension of the network. It cannot be None.
+        output_dim: The output dimension of the network. if None, the last layer would
+            be the last hidden layer.
+        hidden_layers: The sizes of the hidden layers.
+        activation: The activation function to use after each layer.
+    """
+
+    def __init__(self, config: FCConfig):
+        super().__init__()
+        self.input_dim = config.input_dim
+        self.hidden_layers = config.hidden_layers
+
+        activation_class = config.activation
+        self._layers = []
+        self._layers.append(tf.keras.Input(shape=(self.input_dim,)))
+
+        for i in range(len(self.hidden_layers)):
+            self._layers.append(activation_class())
+            self._layers.append(tf.keras.layers.Dense(self.hidden_layers[i]))
+
```

Comment:
For this PR let's focus on just a simple 2-layer fcnet hardcoded inside BCRLModule